### PR TITLE
Add additional functionality to hat stacking

### PR DESCRIPTION
**Pull request by JohnGinnane**
:cl: JohnGinnane
tweak: Hats will fly everywhere if you gib, slip, flip, or spin! You also have a chance to knock hats off of people if you hit them on the head
/:cl:

[why]: """realism""" Also I fixed two syntax errors that somehow compile and run fine.

Videos:

https://gfycat.com/GenuineGlitteringFlies
https://gfycat.com/AssuredZestyIsopod
https://gfycat.com/DishonestHoarseBlueandgoldmackaw
https://gfycat.com/GraySnivelingLeafwing
https://gfycat.com/DescriptiveHardGrison

### CODE REVIEW

**Comment by tortellinitony** [comment]
you can definitely condense all of these into one line procs

**Comment by JohnGinnane** [comment]
Yeah I had the same thought, gonna try that now

**Comment by 2Dollarbill** [comment]
+1 please, my hat addiction.

**Comment by Pyko1** [comment]
should make a chance for X hats to fall if you have more than Y hats and you bump to wall/player.

**Comment by JohnGinnane** [comment]
nah, I wanna keep it somewhat simple and the goal of every shitter should be to have the biggest tower of hats

**Comment by JohnGinnane** [comment]
@tortellinitony that's done

Can someone check this

**Review by ghost — DISMISSED** [review]

**Comment by JohnGinnane** [comment]
@HippieStation/maintainers this is ready

**Comment by ghost** [comment]
hat of hats
+1

File: code/modules/mob/emote.dm [hunk 1]
```diff
@@ -26,6 +26,12 @@
 	if(.)
 		user.SpinAnimation(7,1)
 
+		// Hippie Start - Throw some hats
+		if (istype(user, /mob/living/carbon/))
```
Comment (Carbonhell) [inline code/modules/mob/emote.dm]: couldn't you do this by redefining the proc,calling ..() and then doing this, all in an hippie file

File: code/modules/mob/emote.dm [hunk 2]
```diff
@@ -46,3 +52,9 @@
 					riding_datum.force_dismount(M)
 			else
 				R.unbuckle_all_mobs()
+
+		// Hippie Start - Throw some hats
+		if (istype(user, /mob/living/carbon/))
```
Comment (Carbonhell) [inline code/modules/mob/emote.dm]: same

File: code/modules/mob/emote.dm (hunk 1 above)
Comment (JohnGinnane) [reply code/modules/mob/emote.dm]: Yes

File: code/modules/mob/emote.dm (hunk 2 above)
Comment (JohnGinnane) [reply code/modules/mob/emote.dm]: Yes

**Comment by JohnGinnane** [comment]
@Carbonhell this is ready

**Review by Carbonhell — APPROVED** [review]

**Pull request merged**